File: src/com/company/java_examples/ReadWrite.java
package com.company.java_examples;

import java.util.Arrays;

public class ReadWrite {

    // Leetcode 283

    public static void main(String[] args) {
        int[] nums = {0, 1, 0, 3, 12};
        moveZeroes(nums);
        System.out.println(Arrays.toString(nums));
    }

    public static void moveZeroes(int[] nums) {
        int write = 0;
        for (int read = 0; read < nums.length; read++) {
            if (nums[read] != 0) {
                nums[write] = nums[read];
                write++;
            }
        }
        while (write < nums.length) {
            nums[write] = 0;
            write++;
        }
    }
}

File: src/com/company/kotlin_examples/ReadWrite.kt
package com.company.kotlin_examples

import java.util.*

object ReadWrite {

    // Leetcode 283
    @JvmStatic
    fun main(args: Array<String>) {
        val nums = arrayOf(0, 1, 0, 3, 12)
        moveZeroes(nums)
        println(Arrays.toString(nums))
    }

    fun moveZeroes(nums: Array<Int>) {
        var write = 0
        for (read in nums.indices) {
            if (nums[read] != 0) {
                nums[write] = nums[read]
                write++
            }
        }
        while (write < nums.size) {
            nums[write] = 0
            write++
        }
    }
}